postgres_ffi/src/xlog_utils.rs

```diff
@@ -368,6 +373,7 @@ impl CheckPoint {
     // Next XID should be greater than new_xid.
     // Also take in account 32-bit wrap-around.
     pub fn update_next_xid(&mut self, xid: u32) {
+        let xid = xid.wrapping_add(XID_CHECKPOINT_INTERVAL - 1) & !(XID_CHECKPOINT_INTERVAL - 1);
```

Comment:
You must be careful to not wrap around to values < NormalTransactionId

Reply:
It is handled two lines down, isn't it:
```
        let new_xid = std::cmp::max(xid + 1, pg_constants::FIRST_NORMAL_TRANSACTION_ID);
```

Reply:
Heh, it even says so in the comment. Never mind :-).